foodgroup/auth.go

```diff
@@ -87,7 +90,18 @@ func (s AuthService) RegisterBOSSession(authCookie []byte) (*state.Session, erro
 		return nil, fmt.Errorf("user not found")
 	}
 
-	return s.sessionManager.AddSession(u.DisplayScreenName), nil
+	sess := s.sessionManager.AddSession(u.DisplayScreenName)
+	if sess == nil {
+		return nil, nil
+	}
```

Comment:
I don't think this check really adds any value. If `sess` is nil, then a NPE will be raised from within `AddSession()`.

Reply:
I'll take another look at it. iirc I added this to satisfy a unit test, `TestAuthService_RegisterBOSSession_SessionNotFound`

Reply:
Removed the test in 5241751